src/templates/polykart.ts

```diff
@@ -0,0 +1,118 @@
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import * as fs from 'fs';
+import * as GitHubApi from 'github';
+import * as http from 'http';
+import {Base} from 'yeoman-generator';
+import * as path from 'path';
+
+const gunzip = require('gunzip-maybe');
+const request = require('request');
+const tar = require('tar-fs');
+
+export class PolykartGenerator extends Base {
+
+  _github: GitHubApi;
+  _githubToken: string;
+
+  constructor(args: string | string[], options: any) {
+    super(args, options);
+    this._githubToken = this._getGitHubToken();
+    this._github = new GitHubApi({
+      version: '3.0.0',
+      protocol: 'https',
+    });
+    this._github.authenticate({
+      type: 'oauth',
+      token: this._githubToken,
+    });
+  }
+
+  initializing() {
+    let done = this.async();
+    console.log('Finding latest release of polykart...');
+    return this._getTemplate()
+      .then((release) => {
+        console.log('Latest release URL:', release && release.url);
+        let tarballUrl = release.tarball_url;
+
+        request({
+          url: 'https://api.github.com/repos/PolymerLabs/polykart/tarball/0.1',
+          headers: {
+            'User-Agent': 'request',
+            'Authorization': `token ${this._githubToken}`,
+          }
+        })
+        .on('response', (response) => {
+          if (response.statusCode == 200) {
+            console.log('Downloading template');
+          }
+        })
+        .on('end', () => {
+          console.log('complete');
+          done();
+        })
+        .pipe(gunzip())
+        .pipe(tar.extract(process.cwd(), {
+          ignore: (_, header) => {
+            let splitPath = header.name.split(path.sep);
+            // ignore the top directory in the tarfile to unpack directly to
+            // the cwd
+            return splitPath.length < 2 || splitPath[1] === '';
+          },
+          map: (header) => {
+            let unprefixed = header.name.split(path.sep).slice(1).join(path.sep).trim();
+            // console.log('unpacking', unprefixed);
+            // the ./ is needed to unpack top-level files in the tar, otherwise
+            // they're just not written
+            header.name = './' + unprefixed;
+            return header;
+          }
+        }));
+      })
+      .catch((error) => {
+        console.error('Could not load polykart template');
+        console.error(error);
+        done();
+      });
+  }
+
+  _getGitHubToken() {
+    try {
+      return fs.readFileSync('token', 'utf8').trim();
+    } catch (e) {
+      console.error(`
+You need to create a github token and place it in a file named 'token'.
+The token only needs the 'public repos' permission.
+Generate a token here: https://github.com/settings/tokens
+This restriction will be removed soon!
```

Comment:
This is actually temporary right? Otherwise an env variable would a better choice here.